test/integration-tests/IntegrationTests.Helpers/ProcessHelper.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+
+namespace IntegrationTests.Helpers
+{
+    /// <summary>
+    /// Drains the standard and error output of a process
+    /// </summary>
+    public class ProcessHelper : IDisposable
+    {
+        private readonly ManualResetEventSlim _errorMutex = new();
+        private readonly ManualResetEventSlim _outputMutex = new();
+        private readonly StringBuilder _outputBuffer = new();
+        private readonly StringBuilder _errorBuffer = new();
+
+        public ProcessHelper(Process process)
+        {
+            process.OutputDataReceived += (_, e) => DrainOutput(e, _outputBuffer, _outputMutex);
+            process.ErrorDataReceived += (_, e) => DrainOutput(e, _errorBuffer, _errorMutex);
+
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+        }
+
+        public string StandardOutput => _outputBuffer.ToString();
+
+        public string ErrorOutput => _errorBuffer.ToString();
+
+        public bool Drain(int timeout = Timeout.Infinite)
```

Comment:
How about using `TimeSpan` instead of `int` for `timeout`? Then the caller does not need to think what is the unit.